### PR TITLE
fix incognito unauthorized issue.

### PR DESCRIPTION
**Pull request by sahsisunny**
# Issue - #289 

## What is the change?
- Added a check to see if the user is authenticated or not. If the user is authenticated then only it will return self_user otherwise it will return nothing.
- one file is changed utils.js which is in the root directory of the project.
 
## Is Development Tested?
- [x] Yes

## After Change Screenshots

Screenshot (42)

### CODE REVIEW

**Review by heyrandhir — APPROVED** [review]

File: utils.js [hunk 1]
```diff
@@ -7,7 +7,9 @@ async function getSelfUser() {
     },
   });
   const self_user = await res.json();
-  return self_user;
+  if (res.status === 200) {
+    return self_user;
+  }
```
Comment (RitikJaiswal75) [inline utils.js:12]: What if i get 400 then nothing is sent back also my page goes blank like this

image

**Review by RitikJaiswal75 — CHANGES_REQUESTED** [review]

File: utils.js (hunk 1 above)
Comment (sahsisunny) [reply utils.js]: done

**Review by RitikJaiswal75 — APPROVED** [review]
LGTM

File: utils.js [hunk 2]
```diff
@@ -1,13 +1,17 @@
...
+      credentials: 'include',
+      headers: {
+        'Content-type': 'application/json',
+      },
+    });
+    const self_user = await res.json();
+    if (res.status === 200) {
+      return self_user;
+    }
+  } catch (err) {}
```
Comment (heyrandhir) [inline utils.js:14]: we can console the error here instead of leaving the catch block empty.

**Review by heyrandhir — CHANGES_REQUESTED** [review]

File: utils.js (hunk 2 above)
Comment (sahsisunny) [reply utils.js:14]: it is not necessary to show the error info to user.

**Review by heyrandhir — APPROVED** [review]

**Pull request merged**